internal/server/iigointernal/judiciary.go

```diff
@@ -0,0 +1,191 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+	"github.com/pkg/errors"
+)
+
+type judiciary struct {
+	JudgeID           shared.ClientID
+	budget            shared.Resources
+	presidentSalary   shared.Resources
+	BallotID          int
+	ResAllocID        int
+	speakerID         shared.ClientID
+	presidentID       shared.ClientID
+	EvaluationResults map[shared.ClientID]roles.EvaluationReturn
+	clientJudge       roles.Judge
+}
+
+func (j *judiciary) init() {
+	j.BallotID = 0
+	j.ResAllocID = 0
+}
+
+// returnPresidentSalary returns the salary to the common pool.
+func (j *judiciary) returnPresidentSalary() shared.Resources {
+	x := j.presidentSalary
+	j.presidentSalary = 0
+	return x
+}
+
+// withdrawPresidentSalary withdraws the president's salary from the common pool.
+func (j *judiciary) withdrawPresidentSalary(gameState *gamestate.GameState) bool {
+	var presidentSalary = shared.Resources(rules.VariableMap[rules.PresidentSalary].Values[0])
+	var withdrawAmount, withdrawSuccesful = WithdrawFromCommonPool(presidentSalary, gameState)
+	j.presidentSalary = withdrawAmount
+	return withdrawSuccesful
+}
+
+// sendPresidentSalary sends the president's salary to the president.
+func (j *judiciary) sendPresidentSalary(executiveBranch *executive) {
+	if j.clientJudge != nil {
+		amount, payPresident := j.clientJudge.PayPresident()
+		if payPresident {
+			executiveBranch.budget = amount
+		}
+		return
+	}
+	amount := j.PayPresident()
```

Comment:
wouldn't this override the previous budget allocation. Also this method of checking if the pointer is null should probably be in the other branched sendSalary functions. Just to make sure a client is always authorising the payment

Reply:
Yeah for MVP this is fine, the budgeting system is marked for an overhaul in post-mvp work, I've changed the other branches to share the nil checking